### PR TITLE
implementing search proxy unifiedauth

### LINKED ISSUE #3176: RBAC impersonation isn't working as expected with OIDC integration
**What happened**:
RBAC impersonation isn't working as expected with Okta integration. RBAC policies exist in member cluster and member cluster is in pull mode. Cluster-proxy clusterrole and clusterrolebinding have been updated to include required groups. Requests route through karmada-search component, but don't return expected resources.

**What you expected to happen**:
Have requests from karmada-apiserver respect RBAC policies in the member cluster.

**How to reproduce it (as minimally and precisely as possible)**:
Bind policies to an Okta group on the member cluster and try to validate those policies through the karmada api server.

**Anything else we need to know?**:
We have one member cluster with the ANP proxy in pull mode. Karmada api-server has the following flags for okta.
```
- --oidc-issuer-url=issuer-url
- --oidc-client-id=client-id
- --oidc-username-claim=email
- '--oidc-username-prefix=okta:'
- --oidc-groups-claim=groups
- '--oidc-groups-prefix=okta:'
```

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
```
karmadactl version: version.Info{GitVersion:"v1.4.0-196-gfed2f269-dirty", GitCommit:"fed2f269f5a47e4d37db8bffb29aabe4641112d4", GitTreeState:"dirty", BuildDate:"2023-01-31T16:44:56Z", GoVersion:"go1.19.3", Compiler:"gc", Platform:"darwin/arm64"}
```
- Others:

### PR DESCRIPTION
**Pull request by XiShanYongYe-Chang**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

We can [Proxy Global Resources](https://karmada.io/docs/next/userguide/globalview/proxy-global-resource/) through the `karmada-apiserver`. However, the `search/proxying` does not provide the unified authentication capability like the [cluster/proxy](Unified Authentication), as stated in #3176. 

The current pr supplements this capability.

**Which issue(s) this PR fixes**:
Fixes #3176 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
`karmada-search`: implementing search proxy unifiedauth
```

### CODE REVIEW

File: pkg/controllers/unifiedauth/unified_auth_controller.go [hunk 1]
```diff
@@ -82,64 +84,80 @@ func (c *Controller) Reconcile(ctx context.Context, req controllerruntime.Reques
 }
 
 func (c *Controller) syncImpersonationConfig(cluster *clusterv1alpha1.Cluster) error {
-	// step1: list all clusterroles
-	clusterRoleList := &rbacv1.ClusterRoleList{}
-	if err := c.Client.List(context.TODO(), clusterRoleList); err != nil {
-		klog.Errorf("Failed to list clusterroles, error: %v", err)
+	// step1: found out target RBAC subjects.
```
Comment (RainbowMango) [inline pkg/controllers/unifiedauth/unified_auth_controller.go:87]: `found`
Why use the `past tense` here?

File: pkg/controllers/unifiedauth/unified_auth_controller.go [hunk 2]
```diff
@@ -253,42 +263,58 @@ func (c *Controller) newClusterRoleBindingMapFunc() handler.MapFunc {
...
 		}
 		return c.generateRequestsFromClusterRole(clusterRole)
 	}
 }
 
-// found out which clusters need to sync impersonation config from rules like:
-//
-//	resources: ["cluster/proxy"]
-//	resourceNames: ["cluster1", "cluster2"]
+// generateRequestsFromClusterRole found out which clusters need to sync impersonation config.
```
Comment (RainbowMango) [inline pkg/controllers/unifiedauth/unified_auth_controller.go:273]: `found`

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
> `karmada-search`: implementing search proxy unifiedauth

Please add some information to the release notes.  

`implementing`?

**Comment by XiShanYongYe-Chang** [comment]
> Please add some information to the release notes.
> 
> `implementing`?

Rename it to `support xxx`

**Pull request merged**